src/daily-quest/daily-quest.forms.ts

```diff
@@ -39,10 +39,32 @@ export const DailyQuestForm = z
       path: ["frequency"],
     }
   );
-export type DailyQuestFormType = z.infer<typeof DailyQuestForm>;
-
-export const CreateDailyQuestForm = DailyQuestForm;
 export type CreateDailyQuestFormType = z.infer<typeof CreateDailyQuestForm>;
 
-export const UpdateDailyQuestForm = DailyQuestForm;
+export const UpdateDailyQuestForm = z
+  .object({
+    title: z.string().min(3).max(256),
+    icon: z.string().min(1).max(64),
+    frequency: z
+      .array(z.nativeEnum(Frequency))
+      .min(1, "Select at least one frequency"),
+  })
+  .refine(
+    (data) => {
+      const frequency = data.frequency ?? [];
+      const hasDaily = frequency.includes(Frequency.Daily);
+      const hasWeekdays = frequency.some((frequency) =>
+        weekdaysOnly.includes(frequency)
+      );
+
+      if (hasDaily && hasWeekdays) return false;
+
+      return !(!hasDaily && !hasWeekdays);
```

Comment:
this looks unreadable. can this "!hasDaily && !hasWeekdays" be specified in some human-understandale variable? 
I'm not even sure if it's correct, why do you use ! at the start?

Reply:
This validator ensures that for frequency field we have either `Daily` or one or more of the days:
  Frequency.Mondays,
  Frequency.Tuesdays,
  Frequency.Wednesdays,
  Frequency.Thursdays,
  Frequency.Fridays,
  Frequency.Saturdays,
  Frequency.Sundays,
  
  Cause, it doesnt make sense to have both `Daily` and any of the days set.

Reply:
> I'm not even sure if it's correct, why do you use ! at the start?

There are actually 2 cases here:
1. if doesn't have both daily and weekdays - than fail validation again as the frequency is not optional (return false as `!(true && true)`leads to`false`);
2. if has one of daily or weekdays - than proceed further (return true as both `!(false && true)` and `!(true && false)` leads to `false`);

Reply:
Refactored: 
![image](https://github.com/user-attachments/assets/5f352aea-90f2-4027-a899-cfd5de9755e0)